code_quality.js

```diff
@@ -0,0 +1,211 @@
+const gitChangedFiles = require('git-changed-files');
+const prettier = require('prettier');
+const prettierConfig = require('./prettier_config.json');
+const codeQualityConfig = require('./package.json').codeQuality || {};
+
+const fs = require('fs');
+const { ESLint } = require('eslint');
+const eslint = new ESLint({ fix: true });
+
+const getAllFiles = dir => {
+   let results = [];
+   const list = fs.readdirSync(dir);
+   list.forEach(function (file) {
+      file = dir + '/' + file;
+      const stat = fs.statSync(file);
+      if (
+         file.endsWith('node_modules') ||
+         file.indexOf('.git') !== -1 ||
+         file.indexOf('/dist/') !== -1 ||
+         file.indexOf('\\dist\\') !== -1
+      ) {
+         return [];
+      }
+
+      if (stat && stat.isDirectory()) {
+         /* Recurse into a subdirectory */
+         results = results.concat(getAllFiles(file));
+      } else {
+         /* Is a file */
+         results.push(file);
+      }
+   });
+   return results;
+};
+
+const getChangedFiles = async () => {
+   const committedGitFiles = await gitChangedFiles();
+   return committedGitFiles.unCommittedFiles.filter(file =>
+      fs.existsSync(file) && !file.endsWith('package-lock.json')
+   );
+};
+
+const groupFilesByExtension = files => {
+   return files.reduce((map, file) => {
+      try {
+         if (file.startsWith('./') || file.indexOf('.') === -1) {
+            return map;
+         }
+
+         const extension = file
+            .substring(file.lastIndexOf('.') + 1)
+            .toLowerCase();
+         const array = map[extension] || [];
+         array.push(file);
+         map[extension] = array;
+         return map;
+      } catch (e) {
+         console.error('error --> ', e);
+         return {};
+      }
+   }, {});
+};
+
+const prettyFile = (file, config) => {
+   try {
+      const text = fs.readFileSync(file).toString();
+      const pretty = prettier.format(text, config);
+      if (text === pretty) {
+         return;
+      }
+      console.log('We did prettier the file ' + file);
+      fs.writeFileSync(file, pretty);
+   } catch (error) {
+      console.log('Error in prettier the file ' + file + ': \n' + error);
+   }
+};
+
+const prettyFiles = filesByExtension => {
+   if (!codeQualityConfig.usePrettier) {
+      return; // no need to use pretty;
+   }
+   Object.keys(filesByExtension).forEach(extension => {
+      const files = filesByExtension[extension];
+      const config = prettierConfig[extension];
+      if (!config) {
+         console.log(
+            "We don't have any configuration for extension " +
+               extension +
+               ', please add it in file prettier_config.json'
+         );
+         return;
+      }
+      files.forEach(file => prettyFile(file, config));
+   });
```

Comment:
Could you  console.log "number of prettified files" here as well. It would be pretty helpful.

Reply:
Did it but in methods full and pre_commit:

console.log('Eslint executed in ' + esLintResult.numberFiles + ' files ');